src/schema/session-handler.ts

```diff
@@ -0,0 +1,38 @@
+import gql from 'graphql-tag';
+import get from 'lodash/get';
+import { ZimbraInMemoryCache } from '../apollo/zimbra-in-memory-cache';
+import { ZimbraSessionOptions } from './types';
+
+const SESSIONID_GQL_FRAGMENT = gql`
+	fragment sessionId on Session {
+		id
+	}
+`;
+
+export class SessionHandler {
+	private cache: ZimbraInMemoryCache;
+	private sessionIdKey: string = 'sessionId';
+
+	constructor(options: ZimbraSessionOptions) {
+		this.cache = options.cache;
+	}
+
+	public readSessionId = () => {
+		const sessionIdFragment: any = this.cache.readFragment({
```

Comment:
@silentsakky , Should we use `@client` directive or we are fine with ready and writing fragment directly?